### PR TITLE
Azure AD: Add identityProvider documentation

### PR DESCRIPTION
**Pull request by SaeedAkhter-MSFT**

### CODE REVIEW

File: api-reference/beta/resources/identityprovider.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# identityProvider resource type
+
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Represents an Azure Active Directory Identity Provider.
+
+Examples of Identity Providers are Microsoft Account, Google, Facebook, Amazon, and LinkedIn.  First, each Identity Provider must be configured in an Azure AD tenant or Azure AD B2C tenant.  This usually involves creating an app registration with the Identity Provider in order to get a client id and client secret that can be passed to [create identityProvider](../api/identityprovider_post.md).  Then, each user object in the directory can be federated to any of the tenant's Identity Providers for authentication.  This enables that user login by typing credentials in the Identity Provider's sign in page.  The token from the Identity Provider is validated before the tenant issues a token to the relying party application.
```
Comment (agrabhi) [inline api-reference/beta/resources/identityprovider.md]: The idps have to be configured in their respective systems. For example Google with google and so on

File: api-reference/beta/resources/identityprovider.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# identityProvider resource type
+
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Represents an Azure Active Directory Identity Provider.
```
Comment (agrabhi) [inline api-reference/beta/resources/identityprovider.md]: Will it help to provide use scenarios here?

File: api-reference/beta/resources/identityprovider.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+### Common Properties
+|Property|Type|Required|Nullable|Description|
+|:---------------|:--------|:--------|:--------|:----------|
+|id|String|No|No|The id of the identity provider.|
+|type|String|Yes|No|The identity provider type. It must be one of the following values: <li/>Microsoft<li/>Google<li/>Amazon<li/>LinkedIn<li/>Facebook|
+|name|String|No|No|The display name of the identity provider.|
+|clientId|String|Yes|No|The clientId of the application used to access the identity provider.|
+|clientSecret|String|Yes|No|The client-secret for the app used to access the identity provider. This is write-only. A read operation will return "*****" for security.|
+|scopes|String|No|No|The scopes to request from the identity provider.  This is a space delimited list of scopes.|
```
Comment (agrabhi) [inline api-reference/beta/resources/identityprovider.md]: please remove scopes. Its not part of this release

File: api-reference/beta/resources/identityprovider.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+|[List identityProvider](../api/identityprovider_list.md)|identityProvider collection|List all identityProviders configured in a tenant.|
+
+### Common Properties
+|Property|Type|Required|Nullable|Description|
+|:---------------|:--------|:--------|:--------|:----------|
+|id|String|No|No|The id of the identity provider.|
+|type|String|Yes|No|The identity provider type. It must be one of the following values: <li/>Microsoft<li/>Google<li/>Amazon<li/>LinkedIn<li/>Facebook|
+|name|String|No|No|The display name of the identity provider.|
+|clientId|String|Yes|No|The clientId of the application used to access the identity provider.|
+|clientSecret|String|Yes|No|The client-secret for the app used to access the identity provider. This is write-only. A read operation will return "*****" for security.|
```
Comment (agrabhi) [inline api-reference/beta/resources/identityprovider.md]: please omit "for security "

A read operation will return "*****".

File: api-reference/beta/resources/identityprovider.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+|scopes|String|No|No|The scopes to request from the identity provider.  This is a space delimited list of scopes.|
+
+#### Common Relationships
+|Relationship|Type|Description|
+|:-------------|:-----------|:-----------|
+|appliesTo|[directoryObject](../resources/directoryObject.md) collection|The directory in which the identityProvider is configured.|
```
Comment (agrabhi) [inline api-reference/beta/resources/identityprovider.md]: not sure about this...need to confirm

File: api-reference/beta/resources/identityprovider.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+## JSON representation
+Here is a JSON representation of the resource.
```
Comment (agrabhi) [inline api-reference/beta/resources/identityprovider.md]: is it sample Json representation or the definitions of the object?
In either case, we need alignment

File: api-reference/beta/resources/identityprovider.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
+# identityProvider resource type
```
Comment (agrabhi) [inline api-reference/beta/resources/identityprovider.md]: you need to update the SEO list as well. the navigation pane

File: api-reference/beta/api/identityprovider_delete.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+Delete an existing [identityProvider](../resources/identityprovider.md).
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](../../../concepts/permissions_reference.md).
+
+|Permission type      | Permissions (from least to most privileged)              |
+|:--------------------|:---------------------------------------------------------|
+|Delegated (work or school account) | Directory.AccessAsUser.All IdentityProvider.ReadWrite.All |
```
Comment (agrabhi) [inline api-reference/beta/api/identityprovider_delete.md]: not sure about the Directory.AccessAsUser.All permissions. Please check

File: api-reference/beta/api/identityprovider_get.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+HTTP/1.1 200 OK
+Content-type: application/json
+{
+    "value": [{
+        "id": "Amazon-OAUTH",
+        "type": "Amazon",
+        "name": "displayName",
+        "clientId": "string"
+        "clientSecret": "string",
+        "scopes": "string",
```
Comment (agrabhi) [inline api-reference/beta/api/identityprovider_get.md]: remove scopes

File: api-reference/beta/api/identityprovider_post.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
+# Create identityProvider
+
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Create a new [identityProvider](../resources/identityProvider.md) by specifying display name, identityProvider type, client id, and client secret.  Setting the scopes is optional.
```
Comment (agrabhi) [inline api-reference/beta/api/identityprovider_post.md]: please search for scopes and remove all occurances

File: api-reference/beta/api/identityprovider_post.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+Here is an example of the request.
+
+```http
+POST https://graph.microsoft.com/beta/identityProviders
+Content-type: application/json
+{
+    "name": "Login with Amazon",
+    "type": "Amazon",
+    "clientId": "00000000-0000-0000-0000-000000000000",
+    "clientSecret": "000000000000",
```
Comment (agrabhi) [inline api-reference/beta/api/identityprovider_post.md]: the secret will be "*****"

File: api-reference/beta/api/identityprovider_post.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+Here is an example of the request.
+
+```http
+POST https://graph.microsoft.com/beta/identityProviders
+Content-type: application/json
+{
+    "name": "Login with Amazon",
+    "type": "Amazon",
+    "clientId": "00000000-0000-0000-0000-000000000000",
```
Comment (agrabhi) [inline api-reference/beta/api/identityprovider_post.md]: showing a valid guid will help

File: api-reference/beta/resources/identityprovider.md (hunk 7 above)
Comment (SaeedAkhter-MSFT) [reply api-reference/beta/resources/identityprovider.md]: done

File: api-reference/beta/resources/identityprovider.md (hunk 2 above)
Comment (SaeedAkhter-MSFT) [reply api-reference/beta/resources/identityprovider.md]: done

File: api-reference/beta/api/identityprovider_delete.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+Delete an existing [identityProvider](../resources/identityprovider.md).
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](../../../concepts/permissions_reference.md).
+
+|Permission type      | Permissions (from least to most privileged)              |
+|:--------------------|:---------------------------------------------------------|
+|Delegated (work or school account)|Not supported.|
```
Comment (agrabhi) [inline api-reference/beta/api/identityprovider_delete.md]: IdentityProvider.ReadWrite.All in delegated and app only tokens are not supported

File: api-reference/beta/resources/identityprovider.md [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Represents an Azure Active Directory Identity Provider.
+
+Examples of Identity Providers are Microsoft Account, Google, Facebook, Amazon, and LinkedIn.  First, each Identity Provider must be configured in an Azure AD tenant or Azure AD B2C tenant.  Each identity provider has a process for creating an app registration in their respective system.  For example, creating an app registration with Facebook is done at [developers.facebook.com](https://developers.facebook.com/).  The resulting client id and client secret can be passed to [create identityProvider](../api/identityprovider_post.md).  Then, each user object in the directory can be federated to any of the tenant's Identity Providers for authentication.  This enables the user to login by entering credentials in the Identity Provider's sign in page.  The token from the Identity Provider is validated by Azure AD before the tenant issues a token to the relying party application.
```
Comment (agrabhi) [inline api-reference/beta/resources/identityprovider.md]: please remove -  First, each Identity Provider must be configured in an Azure AD tenant or Azure AD B2C tenant.

**Comment by SaeedAkhter-MSFT** [comment]
#sign-off

**Pull request closed without merging**